src/waltham/waltham-connection.h

```diff
@@ -373,6 +354,83 @@ wth_connection_get_protocol_error(struct wth_connection *conn,
 				  const char **interface,
 				  uint32_t *object_id);
 
+/** Return a registry using the internal wth_display
+ *
+ * \param conn The Waltham connection.
+ * \return The registry obtained.
+ *
+ * The returned registry can then be used to retrieve other global
+ * objects, using its usual interface.
```

Comment:
I think every call creates a new registry object. With Wayland that has been a bit of surprise, so we should at least emphasize it in the documentation if not name the function something like `wth_connection_create_registry()`.

Reply:
This patch is good.